proto/internal/temporal/server/api/persistence/v1/hsm.proto

```diff
@@ -139,3 +139,25 @@ message VersionedTransition {
     int64 transition_count = 2;
 }
 
+message StateMachineTombstoneBatch {
+    // versioned transition the tombstones are created
+    VersionedTransition versioned_transition = 1;
+    repeated StateMachineTombstone state_machine_tombstones = 2; 
+}
+
+message StateMachineTombstone {
+    oneof state_machine_key {
```

Comment:
do we need a type here to indicate the type of StateMachine? Otherwise reading the value could be cumbersome?

Reply:
oneof already contains the type information.
Writing a switch on the type field is basically the same as writing a type switch on the one of field I think. Also additional field means we'll need to check for potential inconsistency between the type field and the oneof field.

There was a similar discussion on this issue [here](https://github.com/temporalio/api/pull/197#discussion_r921100022)